addons/stock_account/models/product.py

```diff
@@ -394,6 +394,14 @@ class ProductCategory(models.Model):
         domain=[('deprecated', '=', False)],
         help="When real-time inventory valuation is enabled on a product, this account will hold the current value of the products.",)
 
+    @api.constrains('property_stock_valuation_account_id', 'property_stock_account_output_categ_id', 'property_stock_account_input_categ_id')
+    def _check_property_stock_valuation_account_id(self):
+        # Prevent user, to set same account number for stock valuation and stock IN/OUT accounts.
+        for category in self:
+            if category.property_stock_valuation_account_id and category.property_stock_valuation_account_id.id in \
+                 [category.property_stock_account_output_categ_id.id, category.property_stock_account_input_categ_id.id]:
+                raise ValidationError(_('The Stock Input and/or Output accounts cannot be the same than the Stock Valuation account.'))
```

Comment:
don't you think something like this would be more readable?
```
        for category in self:
            valuation_account = category.property_stock_valuation_account_id
            input_and_output_accounts = category.property_stock_account_input_categ_id | category.property_stock_account_output_categ_id
            if valuation_account and valuation_account in input_and_output_accounts:
                raise ValidationError(_('The Stock Input and/or Output accounts cannot be the same than the Stock Valuation account.'))
```

i did not really tested it so maybe i'm missing something, can you doucle check?